src/app/games/mancala/ba-awa/tests/BaAwaRules.spec.ts

```diff
@@ -0,0 +1,290 @@
+            // Given a state where a capture-on-the-go is possible for active player !
+            const board: Table<number> = [
+                [0, 0, 0, 0, 0, 8],
+                [3, 2, 0, 0, 1, 0],
+            ];
+            const state: MancalaState = new MancalaState(board, 2, [0, 0]);
+
+            // When performing a move that will create a 4 in a player's house
+            const move: MancalaMove = MancalaMove.of(MancalaDistribution.of(1));
+
+            // Then the capture should be performed
+            const expectedBoard: Table<number> = [
+                [1, 0, 0, 0, 0, 8],
+                [0, 0, 0, 0, 1, 0],
+            ];
+            const expectedState: MancalaState = new MancalaState(expectedBoard, 3, [4, 0]);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+        it('should do multiple capture-on-the-go for opponent when possible', () => {
+            // Given a state where a multiplie capture-on-the-go is possible for passive player !
+            const board: Table<number> = [
+                [3, 3, 0, 0, 0, 8],
+                [5, 1, 0, 0, 0, 0],
+            ];
+            const state: MancalaState = new MancalaState(board, 2, [0, 0]);
+
+            // When performing a move that will capture
+            const move: MancalaMove = MancalaMove.of(MancalaDistribution.of(0));
+
+            // Then the capture should be performed
+            const expectedBoard: Table<number> = [
+                [0, 0, 1, 1, 1, 8],
+                [0, 1, 0, 0, 0, 0],
+            ];
+            const expectedState: MancalaState = new MancalaState(expectedBoard, 3, [0, 8]);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+        it('should do multipe capture-on-the-go for player when possible', () => {
+            // Given a state where a multiple capture-on-the-go is possible for active player !
+            const board: Table<number> = [
+                [0, 0, 0, 0, 0, 8],
+                [3, 3, 3, 0, 0, 1],
+            ];
+            const state: MancalaState = new MancalaState(board, 2, [0, 0]);
+
+            // When performing a move that will create a 4 in a player's house
+            const move: MancalaMove = MancalaMove.of(MancalaDistribution.of(2));
+
+            // Then the capture should be performed
+            const expectedBoard: Table<number> = [
+                [1, 0, 0, 0, 0, 8],
+                [0, 0, 0, 0, 0, 1],
+            ];
+            const expectedState: MancalaState = new MancalaState(expectedBoard, 3, [8, 0]);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+        it(`should mansoon for player zero when total of seed drop to 8 or less (Player.ZERO's turn)`, () => {
+            // Given a state in which Player.ZERO could capture and make total seed drop to zero
```

Comment:
la graine totale ? "make the number of seeds on the board drop to zero" ? ou "remove/capture all seeds from the board"

Reply:
idem juste plus bas au test suivant